unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/HUD/BuilderProjectsPanel/Scripts/Views/ProjectCardView.cs

```diff
@@ -1,52 +1,322 @@
+    [SerializeField] internal Button contextMenuButton;
+    [SerializeField] internal Button expandButton;
+    [SerializeField] internal Button editorButton;
+    
+    [SerializeField] internal RectTransform scenesContainer;
+    [SerializeField] internal VerticalLayoutGroup layoutGroup;
+
+    [SerializeField] private GameObject loadingImgGameObject;
+    [SerializeField] private GameObject publishedGameObject;
+    [SerializeField] private RectTransform downButtonTransform;
+    [SerializeField] internal TextMeshProUGUI projectNameTxt;
+    [SerializeField] internal TextMeshProUGUI projectSizeTxt;
+    [SerializeField] internal TextMeshProUGUI projectSyncTxt;
 
     [Space]
     [SerializeField] private RawImageFillParent thumbnail;
 
-    ProjectData IProjectCardView.projectData => sceneData;
+    ProjectData IProjectCardView.projectData => projectData;
     ISearchInfo IProjectCardView.searchInfo { get; } = new SearchInfo();
     Vector3 IProjectCardView.contextMenuButtonPosition => contextMenuButton.transform.position + CONTEXT_MENU_OFFSET;
 
-    private ProjectData sceneData;
+    private ProjectData projectData;
     private ISearchInfo searchInfo;
     private Vector3 contextMenuButtonPosition;
 
-    public void Setup(ProjectData sceneData) { this.sceneData = sceneData; }
+    private string thumbnailId = null;
+    private Transform defaultParent;
+    private AssetPromise_Texture thumbnailPromise;
+    private bool scenesAreVisible = false;
+    private bool isDestroyed = false;
+    
+    private RectTransform rectTransform;
+
+    private List<Scene> scenesDeployedFromProject = new List<Scene>();
+    private List<IProjectSceneCardView> sceneCardViews = new List<IProjectSceneCardView>();
+
+    private void Awake()
+    {
+        editorButton.onClick.AddListener(EditorButtonClicked);    
+        expandButton.onClick.AddListener(ExpandButtonPressed);    
+        contextMenuButton.onClick.AddListener(EditorButtonClicked);
+
+        rectTransform = GetComponent<RectTransform>();
+    }
+
+    private void OnDestroy()
+    {
+        isDestroyed = true;
+        Dispose();
+    }
+
+    public void Dispose()
+    {
+        AssetPromiseKeeper_Texture.i.Forget(thumbnailPromise);
+        editorButton.onClick.RemoveAllListeners();
+        if(!isDestroyed)
+            Destroy(gameObject);
+    }
+    
+    public void Setup(ProjectData projectData)
+    {
+        this.projectData = projectData;
+        SetName(projectData.title);
+        SetSize(projectData.rows,projectData.colums);
+        
+        SetThumbnail(projectData.id,projectData.thumbnail);
+        
+        ((IProjectCardView)this).searchInfo.SetId(projectData.id);
+    }
+
+    public void SetScenes(List<Scene> scenes)
+    {
+        scenesDeployedFromProject = scenes;
+        publishedGameObject.gameObject.SetActive(true);
+        if (scenes.Count == 0)
+        {
+            syncImage.enabled = false;
+            projectSyncTxt.text = "NOT PUBLISHED";
```

Comment:
Could you create a constant for "NOT PUBLISHED"?

Reply:
Done!